docs: Add release notes for v1.25.0 (#3100)
sqlc-dev/sqlc: docs/reference/changelog.md

# Changelog
All notable changes to this project will be documented in this file.

(v1-25-0)=
## [1.25.0](https://github.com/sqlc-dev/sqlc/releases/tag/v1.25.0)
Released 2024-01-03

### Release notes

#### Add tags to push and verify

> Tags can be added when [pushing](../howto/push.md) schema and queries to [sqlc Cloud](https://dashboard.sqlc.dev). Tags operate like git tags, meaning you can overwrite previously published tag values. We suggest tagging pushes to associate them with git tags or git branches.

Reviewer: ```suggestion
You can add tags when [pushing](../howto/push.md) schema and queries to [sqlc Cloud](https://dashboard.sqlc.dev). Tags operate like git tags, meaning you can overwrite previously-pushed tag values. We suggest tagging pushes to associate them with something relevant from your environment, e.g. a git tag or branch name.
```

```

> $ sql push --tags v1.0.0

Reviewer: ```suggestion
$ sqlc push --tag v1.0.0
```

```

Once you've created a tag, you can refer to it when [verifying](../howto/verify.md) changes, allowing you
to compare the existing schema against a known set of previous queries.

```

> $ sql verify --against v1.0.0

Reviewer: ```suggestion
$ sqlc verify --against v1.0.0
```

```

> #### See ya, `cgo`

Reviewer: ```suggestion
#### C-ya, `cgo`
```

Over the last month, we've switched out a few different modules to remove our reliance on [cgo](https://go.dev/blog/cgo). Previously, we needed cgo for three separate functions:

- Parsing PostgreSQL queries with [pganalyze/pg_query_go](https://github.com/pganalyze/pg_query_go)
- Running SQLite databases with [mattn/go-sqlite3](https://github.com/mattn/go-sqlite3)
- Executing WASM / WASI code with [bytecodealliance/wasmtime-go](https://github.com/bytecodealliance/wasmtime-go)

> With the help of the community, we found cgo-less alternatives for each module:

Reviewer: ```suggestion
With the help of the community, we found cgo-free alternatives for each module:
```
